byway/src/main/java/com/google/sps/servlets/DestinationsServlet.java

```diff
@@ -0,0 +1,74 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns start location and destinations user inputs */
+@MultipartConfig
+@WebServlet("/api/destinations")
+public class DestinationsServlet extends HttpServlet {
+
+private final UserLocations places = new UserLocations("", new ArrayList<String>());
+private final Gson gson = new Gson();
+private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+private Key userKey;
+
+  @Override
+  public void init() {
+    Entity userEntity = new Entity("UserInputs");
+    userEntity.setProperty("start", "");
+    userEntity.setProperty("destinations", new ArrayList<String>());
+    datastore.put(userEntity);
+    userKey= userEntity.getKey();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Entity entity; 
+    try{ 
+      entity = datastore.get(userKey);
+      String start = (String) entity.getProperty("start");
+      ArrayList<String> destinations = (ArrayList<String>) entity.getProperty("destinations");
+      UserLocations userLocations = new UserLocations(start, destinations); 
+      System.out.println(userLocations.getDestinations());   
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(userLocations));
+    } catch(EntityNotFoundException e){
+      System.out.println("error");
+    } 
```

Comment:
Try to scope catch blocks to be as narrow as you can, so that they can't accidentally catch errors from parts of the code that you didn't intend. Note that you can return from a `catch` block if there's no sensible default; though in this case I would consider adding an appropriate HTTP error code to the response, rather than just leaving the `response` empty.

For logging mesages, have a look at https://google.github.io/flogger/. I've already added a dependency on this to pom.xml, so you should be able to just import it, create a static logger in the top of this calss, and change println for `logger.atError().withCause(e).log("Unable to find UserLocations Entity %s", userKey);`

Reply:
added